lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -459,10 +478,31 @@ data TransactionInfo = TransactionInfo
     -- ^ Creation time of the block including this transaction.
     , txInfoMetadata :: !(Maybe TxMetadata)
     -- ^ Application-specific extension data.
+    , txInfoIsValidScript :: !ScriptValidation
+    -- ^ Tag indicating whether non-native scripts in this transaction are
+    -- expected to validate. This is added by the block creator when
+    -- constructing the block. For pre-Alonzo era transactions, scripts are not
+    -- supported, and so script validation always passes.
     } deriving (Generic, Show, Eq)
 
 instance NFData TransactionInfo
 
+-- | Indicates whether the script associated with a transaction has passed or
+-- failed validation. Pre-Alonzo era, scripts were not supported.
+data ScriptValidation = ScriptsNotSupported
+                      | ScriptValidationPassed
+                      | ScriptValidationFailed
+  deriving (Generic, Show, Eq, Ord)
```

Comment:
Suggestions:
1. Use the `Tx` prefix, since this relates to a transaction and appears in the `Tx` module.
2. Use `Validity`, as we can then say "what is the validity of the script for this transaction"?

As for the second point, we could use "validation status".

```suggestion
-- | Indicates whether the script associated with a transaction has passed or
-- failed validation. Pre-Alonzo era, scripts were not supported.
data TxScriptValidity
    = TxScriptsUnsupported
    -- ^ Indicates that scripts were not supported in the era in which this
    -- transaction was created.
    | TxScriptValid
    -- ^ Indicates that the script passed validation.
    | TxScriptInvalid
    -- ^ Indicates that the script failed validation.
  deriving (Generic, Show, Eq, Ord)
```

Reply:
ee0aaf9